File: src/Domain/Entities/Transaction.cs
using System;
using System.Collections.Generic;
using Domain.Common;
using Domain.Enums;

namespace Domain.Entities
{
    public class Transaction : BaseEntity, IHasDomainEvent
    {
        public Guid Id { get; init; }
        public Guid AssignedSupervisorId { get; set; }
        
        public Guid AskId { get; set; }
        public Ask Ask { get; set; }
        
        public Guid BidId { get; set; }
        public Bid Bid { get; set; }
        
        public TransactionStatus Status { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal SellerFee { get; set; }
        public decimal BuyerFee { get; set; }
        public decimal Payout { get; set; }

        public List<DomainEvent> DomainEvents { get; set; }
    }
}

File: src/Client/src/app/supervisor-panel/transaction.ts
import { Ask } from "../shared/ask.model";
import { Bid } from "../shared/bid.model";

export class Transaction {
    constructor(
        public id: string,
        public assignedSupervisorId: string,
        public companyProfit: string,
        public ask: Ask,
        public bid: Bid,
        public buyerFee: number,
        public totalBuyerCost: number,
        public status: string,
        public startDate: Date,
        public endDate: Date
    ) {}
}